### PR TITLE
MOTECH-2928 Added Option to Hide Task Bubbles

### PR DESCRIPTION
**Pull request by jbanasiewicz**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: modules/tasks/tasks/src/main/resources/webapp/js/controllers.js [hunk 1]
```diff
@@ -260,6 +260,11 @@
     controllers.controller('TasksManageCtrl', function ($scope, ManageTaskUtils, Channels, DataSources, Tasks, Triggers,
                 $q, $timeout, $stateParams, $http, $filter, ModalFactory, LoadingModal, HelpStringManipulation) {
 
+        $scope.showBubbles = true;
+        $scope.bubbleFunction = function() {
```
Comment (pkornowski) [inline modules/tasks/tasks/src/main/resources/webapp/js/controllers.js]: Name of this function is not intuitive. Please change it to showHideBubbles for example.

**Comment by pkornowski** [comment]
@kmadej 
Looks good to me.

File: modules/tasks/tasks/src/main/resources/webapp/partials/form.html [hunk 2]
```diff
@@ -1,6 +1,14 @@
 <div class="row create-edit-task" data-ng-cloak>
-    <h4 ng-hide="task.id" class="frame-title"><span ng-bind="msg('task.section.newTask')"></span> <a href="#/tasks/dashboard" type="button" class="btn btn-sm btn-danger btn-in-frame-header pull-right"><i class="fa fa-times"></i></a></h4>
-    <h4 ng-show="task.id" class="frame-title"><span ng-bind="msg('task.section.editTask')"></span> <a href="#/tasks/dashboard" type="button" class="btn btn-sm btn-primary btn-in-frame-header pull-right"><i class="fa fa-level-up"></i></a></h4>
+    <h4 ng-hide="task.id" class="frame-title"><span ng-bind="msg('task.section.newTask')"></span>
+        <a href="#/tasks/dashboard" type="button" class="btn btn-sm btn-danger btn-in-frame-header pull-right"><i class="fa fa-times"></i></a>
+        <button ng-hide="showBubbles" type="button" class="btn btn-sm btn-in-frame-header btn-success btn-show-hide-bubbles" ng-click="showHideBubbles()">Show Value Bubbles</button>
+        <button ng-show="showBubbles" type="button" class="btn btn-sm btn-in-frame-header btn-danger btn-show-hide-bubbles" ng-click="showHideBubbles()">Hide Value Bubbles</button>
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form.html]: use properties, it should be translatable

File: modules/tasks/tasks/src/main/resources/webapp/partials/form.html [hunk 3]
```diff
@@ -1,6 +1,14 @@
...
+    </h4>
+    <h4 ng-show="task.id" class="frame-title"><span ng-bind="msg('task.section.editTask')"></span>
+        <a href="#/tasks/dashboard" type="button" class="btn btn-sm btn-primary btn-in-frame-header pull-right"><i class="fa fa-level-up"></i></a>
+        <button ng-hide="showBubbles" type="button" class="btn btn-sm btn-in-frame-header btn-success btn-show-hide-bubbles" ng-click="showHideBubbles()">Show Value Bubbles</button>
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form.html]: use properties, it should be translatable

File: modules/tasks/tasks/src/main/resources/webapp/partials/form.html [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
+        <button ng-show="showBubbles" type="button" class="btn btn-sm btn-in-frame-header btn-danger btn-show-hide-bubbles" ng-click="showHideBubbles()">Hide Value Bubbles</button>
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form.html]: use properties, it should be translatable

File: modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties [hunk 5]
```diff
@@ -78,6 +78,8 @@ task.section.save=Save
 task.section.selectEvents=Create a Task to push data from...
 task.section.taskDescription=Task Description
 task.section.recentActivityForTask=Recent Activity for Task
+task.section.showBubbles=Show Vlaue Bubbles
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/messages/messages.properties]: typo, should be 'Value'

File: modules/tasks/tasks/src/main/resources/webapp/partials/form-data-source.html [hunk 6]
```diff
@@ -106,7 +106,7 @@
                     <a target="_self" href="#helpModalFieldSyntaxWithBrackets" role="button" class="btn btn-primary pull-left" data-toggle="modal">{{msg('task.help')}}</a>
                 </div>
             </div>
-            <div class="form-group margin-before3 control-section" ng-hide="step.displayName == undefined">
+            <div class="form-group margin-before3 control-section" ng-if="step.displayName != undefined" ng-show="showBubbles">
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/partials/form-data-source.html]: why do you use here ng-if and ng-show, maybe try to join this conditions

File: modules/tasks/tasks/src/main/resources/webapp/js/controllers.js [hunk 7]
```diff
@@ -260,6 +260,11 @@
     controllers.controller('TasksManageCtrl', function ($scope, ManageTaskUtils, Channels, DataSources, Tasks, Triggers,
                 $q, $timeout, $stateParams, $http, $filter, ModalFactory, LoadingModal, HelpStringManipulation) {
 
+        $scope.showBubbles = true;
+        $scope.showHideBubbles = function() {
```
Comment (kmadej) [inline modules/tasks/tasks/src/main/resources/webapp/js/controllers.js]: showHiddenBubbles

**Pull request merged**